build.xml

```diff
@@ -1014,28 +1107,45 @@ cd ..
       <arg value="JNA_JNI_VERSION=${jni.version}"/>
       <arg value="CHECKSUM=${jni.md5}"/>
     </exec>
+    <!-- Include only dynamic library variants for JVM JAR targets. -->
     <mkdir dir="${classes}/${native.path}"/>
     <copy todir="${classes}/${native.path}">
       <fileset dir="${build.native}"
-               includes="jnidispatch.dll,libjnidispatch.*"/>
+               includes="jnidispatch.dll,libjnidispatch.*"
+                excludes="*.lib,*.a"/>
     </copy>
     <mkdir dir="${eclipse.classes}/${native.path}"/>
     <copy todir="${eclipse.classes}/${native.path}"
           failonerror="false">
+      <fileset dir="${build.native}"
+               includes="jnidispatch.dll,libjnidispatch.*"
+               excludes="*.lib,*.a"/>
+    </copy>
+    <!-- Include both dynamic and static library variants for all operating systems in the GraalVM JAR. -->
+    <mkdir dir="${classes-graalvm}/${native.path}"/>
+    <copy todir="${classes-graalvm}/${native.path}">
       <fileset dir="${build.native}"
                includes="jnidispatch.dll,libjnidispatch.*"/>
     </copy>
     <!-- For web start, native libraries may be provided in the root of -->
     <!-- an included jar file -->
     <jar jarfile="${build}/${native.jar}" createUnicodeExtraFields="never" encoding="UTF-8">
-      <fileset dir="${build.native}" includes="jnidispatch.dll,libjnidispatch.*"/>
+      <fileset dir="${build.native}" includes="jnidispatch.dll,libjnidispatch.*" excludes="*.a,*.lib"/>
```

Comment:
current JARs (non-static) do not ship the static library, to avoid duplicating lib size contribution to JARs